File: MyStoryAppBackendPy/app/api/image.py
"""
图片扩展 API
"""

from fastapi import APIRouter, UploadFile, File, Form, HTTPException
from typing import Optional

from app.models.schemas import ImageExpandResponse
from app.services.bailian_image import bailian_image_service
from app.core.logger import logger

router = APIRouter()


@router.post("/expand", response_model=ImageExpandResponse)
async def expand_image(
    image: UploadFile = File(...),
    style: str = Form(default="cinematic")
):
    """
    扩展图片为 16:9 宽屏
    
    - **image**: 图片文件
    - **style**: 扩展风格 (cinematic/anime/realistic/dreamy/vintage/artistic)
    """
    # 验证图片格式
    if not bailian_image_service.validate_image(image.content_type):
        raise HTTPException(status_code=400, detail="不支持的图片格式")
    
    try:
        # 保存上传的图片
        import uuid
        from pathlib import Path
        from app.config import settings
        
        upload_dir = Path(settings.UPLOAD_DIR)
        upload_dir.mkdir(parents=True, exist_ok=True)
        
        file_ext = image.filename.split(".")[-1] if "." in image.filename else "jpg"
        file_path = upload_dir / f"{uuid.uuid4()}.{file_ext}"
        
        content = await image.read()
        file_path.write_bytes(content)
        
        # 调用百炼扩展图片
        expanded_url = await bailian_image_service.expand_image(str(file_path), style)
        
        return ImageExpandResponse(
            success=True,
            data={
                "task_id": str(uuid.uuid4()),
                "expanded_image_url": expanded_url
            }
        )
        
    except Exception as e:
        logger.exception("图片扩展失败")
        raise HTTPException(status_code=500, detail=str(e))


@router.post("/validate")
async def validate_image_format(mimetype: str):
    """验证图片格式是否支持"""
    is_valid = bailian_image_service.validate_image(mimetype)
    return {"success": True, "data": {"valid": is_valid}}

File: MyStoryAppBackend/src/routes/image.ts
import { Router } from 'express';
import multer from 'multer';
import path from 'path';
import { v4 as uuidv4 } from 'uuid';
import { body } from 'express-validator';
import { config } from '../config';
import { imageService } from '../services/imageService';
import { taskService } from '../services/taskService';
import { asyncHandler } from '../middleware/errorHandler';
import { logger } from '../utils/logger';

const router = Router();

// 文件上传配置
const storage = multer.diskStorage({
  destination: (req, file, cb) => {
    cb(null, config.storage.local.uploadDir);
  },
  filename: (req, file, cb) => {
    const uniqueName = `${uuidv4()}${path.extname(file.originalname)}`;
    cb(null, uniqueName);
  }
});

const upload = multer({
  storage,
  limits: { fileSize: 10 * 1024 * 1024 }, // 10MB
  fileFilter: (req, file, cb) => {
    if (imageService.validateImage(file.mimetype)) {
      cb(null, true);
    } else {
      cb(new Error('Invalid file type. Only images are allowed.'));
    }
  }
});

/**
 * POST /api/v1/image/expand
 * 扩展图片
 */
router.post(
  '/expand',
  upload.single('image'),
  body('style').optional().isString(),
  asyncHandler(async (req, res) => {
    if (!req.file) {
      return res.status(400).json({
        success: false,
        error: { message: 'No image file provided' }
      });
    }
    
    const style = req.body.style || 'cinematic';
    const taskId = uuidv4();
    
    // 异步处理
    const processImage = async () => {
      try {
        const expandedUrl = await imageService.expandImage(req.file!.path, style);
        return { taskId, expandedImageUrl: expandedUrl };
      } catch (error) {
        logger.error('Image expansion error:', error);
        throw error;
      }
    };
    
    const result = await processImage();
    
    res.json({
      success: true,
      data: result
    });
  })
);

/**
 * POST /api/v1/image/validate
 * 验证图片格式
 */
router.post(
  '/validate',
  asyncHandler(async (req, res) => {
    const { mimetype } = req.body;
    const isValid = imageService.validateImage(mimetype);
    
    res.json({
      success: true,
      data: { valid: isValid }
    });
  })
);

export default router;
